### DISCUSSION TITLE
Storage Policy To Limit File size and types

### DISCUSSION THREAD

**Discussion by rlee1990** [supabase/supabase#7877] [Questions] [answered]
First is there a list of all the SQL policy commands that are allowed?
Second is there a way to know the size of a file trying to be uploaded and set a limit?
Third is there a way to set a limit for which type of files can be uploaded? I saw a template for limiting to jpg but could I include more types with say just a comma or is there more to it?

**Comment by GaryAustin1** [comment, marked as answer]
Policies are an just an extra SQL where statement, so quite broad.

You can only set max file size for all files:
![image](https://user-images.githubusercontent.com/54564956/180672818-31d55faf-2d32-4a0c-ab54-19065190581d.png)

You can certainly add to the types of files using Postrest SQL with an in statement.  See last part of this: https://github.com/supabase/supabase/discussions/2001

Edit: Discussion of adding a per bucket size in the future: https://github.com/supabase/storage-api/issues/134

**Reply by rlee1990** [reply]
@GaryAustin1 My question might have been a bit broad. What I wanted to do was have say 2 buckets 1 is a paid bucket the other a free-use bucket. For the free-use bucket, I wanted to check when the user tries to upload the file, and if it's over say 25MB then not allow it. Where the paid bucket would not have this restriction. If this is not possible that's fine. Or if I would have to allow it to be uploaded and then have a function check the size limit I can do that.

**Reply by GaryAustin1** [reply]
There is nothing server side to see the file size.  
I added an edit, which if implemented, is probably what you want.  https://github.com/supabase/storage-api/issues/134

**Reply by rlee1990** [reply]
Thanks @GaryAustin1 still new to the SQL side trying to learn as I go.
